File: Python/game.py
#Imports
import random
import os
import time

#Variables
playerOne = ""
playerOneCards = 0
playerOneDeck = []
playerTwo = ""
playerTwoCards = 0
playerTwoDeck = []
turn = 1
currentRound = 1
redDeck = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
blackDeck = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
yellowDeck = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
total = len(redDeck) + len(blackDeck) + len(yellowDeck)

_pOneC = ""
_pOneN = -1
_pTwoC = ""
_pTwoN = -1

#Functions
def clear():
    os.system("cls")


def roundWinner(cardOne, cardTwo):
    if cardOne == "Red" and cardTwo == "Black":
        return 1
    elif cardOne == "Yellow" and cardTwo == "Red":
        return 1
    elif cardOne == "Black" and cardTwo == "Yellow":
        return 1
    else:
        return 2

def deckFromColour(colour):
    if colour == "Red":
        return redDeck
    elif colour == "Black":
        return blackDeck
    elif colour == "Yellow":
        return yellowDeck

def updateNewDeck(colour, deck):
    if colour == "Red":
        global redDeck
        redDeck = deck
    elif colour == "Black":
        global blackDeck
        blackDeck = deck
    elif colour == "Yellow":
        global yellowDeck
        yellowDeck = deck

def updateTotal():
    global total
    total = len(redDeck) + len(blackDeck) + len(yellowDeck)

def getPlayerName(id):
    if id == 1: return playerOne
    elif id == 2: return playerTwo
    else: return None

def cardsFromId(id):
    if id == 1: return playerOneCards
    elif id == 2: return playerTwoCards
    else: return None

def updateTempValue(id, colour, num):
    if id == 1:
        global _pOneC
        global _pOneN
        _pOneC = colour
        _pOneN = num
    elif id == 2:
        global _pTwoC
        global _pTwoN
        _pTwoC = colour
        _pTwoN = num

def updatePlayerDeck(id, colour, num):
    if id == 1:
        global playerOneDeck
        playerOneDeck.append("" + colour + " " + str(num))
    elif id == 2:
        global playerTwoDeck
        playerTwoDeck.append("" + colour + " " + str(num))

def deckFromId(id):
    if id == 1: return playerOneDeck
    elif id == 2: return playerTwoDeck
    else: return None

def clearTempValues():
    global _pOneC
    global _pOneN
    global _pTwoC
    global _pTwoN

    _pOneC = ""
    _pOneN = -1
    _pTwoC = ""
    _pTwoN = -1


#Start up
print("""
That one Card Game: Python Edition!

Welcome! I guess let's just get into the sauce, eh?
First off, I'm gunna need the names of the 2 people playing.
Help me out and enter your names below!
""")
playerOne = input("Player 1's Name > ")
playerTwo = input("Player 2's Name > ")

clear()

print("Hello, " + playerOne + " and " + playerTwo + "! Nice to meet you. Now, let's get started.\n")
print("""
Rules

Here are the rules:
- Both players take a card from the deck.
- If both players have the same colour, the player with the highest number wins the round.
- Else, a winner will be decided based on this table:

Card      Card      Winner
Red       Black     Red
Yellow    Red       Yellow
Black     Yellow    Black

- The winning player of that round will keep both of the cards.
- The game will end when there are no cards left.

(Note that in the 3 decks there are 10 cards each)

Press enter to start!
""")
input("")
clear()

#The game
while total > 0:
    print("Round " + str(currentRound))
    print(getPlayerName(turn) + "'s Turn")
    print("Drawing...")

    drawColour = ""
    drawNumber = -1
    _deck = []

    while len(_deck) == 0:
        _rand = random.randint(1, 3)
        if _rand == 1:
            drawColour = "Red"
        elif _rand == 2:
            drawColour = "Black"
        elif _rand == 3:
            drawColour = "Yellow"

        _deck = deckFromColour(drawColour)

    drawNumber = random.choice(_deck)
    _deck.remove(drawNumber)
    updateNewDeck(drawColour, _deck)
    updateTotal()
    updateTempValue(turn, drawColour, drawNumber)
    updatePlayerDeck(turn, drawColour, drawNumber)

    time.sleep(1)

    print(getPlayerName(turn) + " drew a " + drawColour + " " + str(drawNumber))

    input("\nPress enter to continue...\n")
    if turn == 2: #TODO: tell the player their cards
        clear()
        turn = 1
        _roundWinner = ""
        if _pOneC == _pTwoC:
            if _pOneN == _pTwoN:
                _roundWinner = "Everyone"
                playerOneCards += 1
                playerTwoCards += 1
            elif _pOneN > _pTwoN:
                playerOneCards += 2
                _roundWinner = playerOne
            else:
                playerTwoCards += 2
                _roundWinner = playerTwo
        else:
            _winner = roundWinner(_pOneC, _pTwoC)
            if _winner == 1:
                playerOneCards += 2
                _roundWinner = playerOne
            else:
                playerTwoCards += 2
                _roundWinner = playerTwo

        if total <= 0:
            print("Calculating final scores! One sec...")
            time.sleep(3)
            input("Done. Press enter to see the winner!")
        else:
            print("""
End of Round """ + str(currentRound) + """ Results:
        
""" + _roundWinner + """ has won the round!
        
Scores:
""" + playerOne + """ > """ + str(playerOneCards) + """ cards
""" + playerTwo + """ > """ + str(playerTwoCards) + """ cards

There are """ + str(total) + """ cards left.
""")
            input("\nPress enter to go to the next round!")
            clearTempValues()
            currentRound += 1
            clear()
    else:
        print(" ")
        turn = 2

# Announcing the winner
_globalWinner = ""
_globalWinnerId = -1
_globalWinnerCards = -1
_globalLooserId = -1
_boolDraw = False

if playerOneCards > playerTwoCards:
    _globalWinner = playerOne
    _globalWinnerCards = playerOneCards
    _globalWinnerId = 1
    _globalLooserId = 2
elif playerOneCards < playerTwoCards:
    _globalWinner = playerTwo
    _globalWinnerCards = playerTwoCards
    _globalWinnerId = 2
    _globalLooserId = 1
else:
    _boolDraw = True
    _globalWinnerCards = playerOneCards # they are both the same, so it doesn't matter which one I use

clear()
if _boolDraw == False:
    print("""
Game. OVER!

""" + _globalWinner + """ has won the game!
They had """ + str(_globalWinnerCards) + """ cards.

""" + getPlayerName(_globalLooserId) + """ was """ + str((_globalWinnerCards - cardsFromId(_globalLooserId)) + 1) + """ cards away from winning!
They had """ + str(cardsFromId(_globalLooserId)) + """ cards.

Thanks for playing!
    """)

    _winnerDeckText = ""
    for i in range(len(deckFromId(_globalWinnerId))):
        if _winnerDeckText == "":
            _winnerDeckText += deckFromId(_globalWinnerId)[i]
        else:
            _winnerDeckText += ", " + deckFromId(_globalWinnerId)[i]
    print("Winner's Deck > " + _winnerDeckText)

    input("")
else:
    print("""
    Game. OVER!

    It was a tie! Both player had """ + str(_globalWinnerCards) + """ cards!

    Thanks for playing!
        """)
    input("")

File: Ruby/Game.rb
# Variables
$playerone = ""
$playerone_cards = 0
$playerone_deck = []

$playertwo = ""
$playertwo_cards = 0
$playertwo_deck = []

$turn = 1
$currentround = 1

$red_deck = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
$black_deck = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
$yellow_deck = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
$total = $red_deck.size + $black_deck.size + $yellow_deck.size

$_pone_c = ""
$_pone_n = -1
$_ptwo_c = ""
$_ptwo_n = -1

#Functions
def clear
  system("cls")
end

def round_winner(cardone, cardtwo)
  if cardone.eql? "Red" and cardtwo.eql? "Black"
    1
  elsif cardone.eql? "Yellow" and cardtwo.eql? "Red"
    1
  elsif cardone.eql? "Black" and cardtwo.eql? "Yellow"
    1
  else
    2
  end
end

def deck_from_colour(colour)
  case colour
  when "Red" then
    return $red_deck
  when "Black" then
    return $black_deck
  when "Yellow" then
    return $yellow_deck
  else
    return nil
  end
end

def update_new_deck(colour, deck)
  case colour
  when "Red" then
    $red_deck = deck
  when "Black" then
    $black_deck = deck
  when "Yellow" then
    $yellow_deck = deck
  else
    return
  end
end

def update_total()
  $total = $red_deck.size + $black_deck.size + $yellow_deck.size
end

def get_name_from_id(id)
  case id
  when 1 then
    return $playerone
  when 2 then
    return $playertwo
  else
    return nil
  end
end

def get_cards_from_id(id)
  case id
  when 1 then
    return $playerone_cards
  when 2 then
    return $playertwo_cards
  else
    return nil
  end
end

def update_temp_values(id, colour, num)
  case id
  when 1 then
    $_pone_c = colour
    $_pone_n = num
  when 2 then
    $_ptwo_c = colour
    $_ptwo_n = num
  else
    return
  end
end

def update_player_deck(id, colour, num)
  case id
  when 1 then
    $playerone_deck.append("" + colour + " " + num.to_s)
  when 2 then
    $playertwo_deck.append("" + colour + " " + num.to_s)
  else
    return
  end
end

def get_deck_from_id(id)
  case id
  when 1 then
    return $playerone_deck
  when 2 then
    return $playertwo_deck
  else
    return nil
  end
end

def clear_temp_values()
  $_pone_c = ""
  $_pone_n = -1
  $_ptwo_c = ""
  $_ptwo_n = -1
end

#Start up
clear
print "That one Card Game: Ruby Eddition!\n\n" +
          "Welcome! I guess let's just get into the sauce, eh?\n" +
          "First off, I'm gunna need the names of the 2 people playing.\n" +
          "Help me out and enter your names below!\n"

print "\nPlayer 1's Name > "
$playerone = gets.chop
print "\nPlayer 2's Name > "
$playertwo = gets.chop
clear

puts "Hello, " + $playerone + " and " + $playertwo + "! Nice to meet you. Now, let's get started."
puts "\nRules\n\n" +
         "Here are the rules:\n" +
         "- Both players take a card from the deck.\n" +
         "- If both players have the same colour, the player with the highest number wins the round.\n" +
         "- Else, a winner will be deided based on this table:\n\n" +
         "Card      Card      Winner\n" +
         "Red       Black     Red\n" +
         "Yellow    Red       Yellow\n" +
         "Black     Yellow    Black\n\n" +
         "- The winning player of that round will keep both of the cards.\n" +
         "- The game will end when there are no cards left.\n\n" +
         "(Note that in the 3 decks there are 10 cards each)\n\n" +
         "Press enter to start!"
gets.chop
clear

while $total > 0 do
  puts "Round " + $currentround.to_s
  puts get_name_from_id($turn) + "'s Turn"
  puts "Drawing...\n"

  draw_colour = ""
  draw_number = -1
  deck = []

  while deck.size == 0 do
    rand_deck = rand(1..3)

    if rand_deck == 1 then
      draw_colour = "Red"
    elsif rand_deck == 2 then
      draw_colour = "Black"
    elsif rand_deck == 3 then
      draw_colour = "Yellow"
    end

    deck = deck_from_colour(draw_colour)
  end

  draw_number = deck.sample
  deck.delete(draw_number)
  update_new_deck(draw_colour, deck)
  update_total
  update_temp_values($turn, draw_colour, draw_number)
  update_player_deck($turn, draw_colour, draw_number)

  sleep(1)

  puts "\n" + get_name_from_id($turn) + " drew a " + draw_colour + " " + draw_number.to_s

  puts "Press enter to continue..."
  gets.chop

  if $turn == 2
    clear
    $turn = 1

    round_winner = ""

    if $_pone_c == $_ptwo_c
      if $_pone_n == $_ptwo_n
        round_winner = "Everyone"
        $playerone_cards += 1
        $playertwo_cards += 1
      elsif $_pone_n > $_ptwo_n then
        $playerone_cards += 2
        round_winner = $playerone
      else
        $playertwo_cards += 2
        round_winner = $playertwo
      end
    else
      winner = round_winner($_pone_c, $_ptwo_c)
      if winner == 1
        $playerone_cards += 2
        round_winner = $playerone
      else
        $playertwo_cards += 2
        round_winner = $playertwo
      end
    end

    if $total <= 0
      puts "Calculating final scores! One sec..."
      sleep(3)
      puts "Done. Press enter to see the winner!"
      gets.chomp
    else
      puts "End of Round " + $turn.to_s + " Results:\n\n" +
               round_winner + " has won the round!\n\n" +
               "Scores:\n" +
               "" + $playerone + " > " + $playerone_cards.to_s + " cards\n" +
               "" + $playertwo + " > " + $playertwo_cards.to_s + " cards\n\n" +
               "There are " + $total.to_s + " cards left."
      puts "Press enter to go to the next round!"
      gets.chop
      clear_temp_values
      $currentround += 1
      clear
    end
  else
    puts " "
    $turn = 2
  end
end

# Announcing the winner

_winner = ""
_winner_id = -1
_winner_cards = -1
_looser_id = -1
_draw = false

if $playerone_cards > $playertwo_cards
  _winner = $playerone
  _winner_id = 1
  _winner_cards = $playerone_cards
  _looser_id = 2
elsif $playerone_cards < $playertwo_cards
  _winner = $playertwo
  _winner_id = 2
  _winner_cards = $playertwo_cards
  _looser_id = 1
else
  _draw = true
  _winner_cards = $playerone_cards # They are both the same, so it doesn't matter
end

clear
if _draw
  puts "Game. OVER!\n\n" +
           "It was a tie! Both players had " + _winner_cards.to_s + " cards!\n\n" +
           "Thanks for playing!"

  gets.chop
else
  puts "Game. OVER!\n\n" +
           "" + _winner + " has won the game!\n" +
           "They had " + _winner_cards.to_s + " cards.\n\n" +
           "" + get_name_from_id(_looser_id) + " was " + ((_winner_cards - get_cards_from_id(_looser_id)) + 1).to_s + " cards away from winning!\n" +
           "They had " + get_cards_from_id(_looser_id).to_s + " cards.\n\n" +
           "Thanks for playing!"

  _winner_text = ""
  get_deck_from_id(_winner_id).each do |i|
    if _winner_text == ""
      _winner_text += i
    else
      _winner_text += ", " + i
    end
  end
  print "\nWinner's Deck > " + _winner_text + "\n"

  gets.chop
end